crates/astria-sequencer-validation/src/proof.rs

```diff
@@ -26,12 +26,12 @@ use sha2::Sha256;
 /// inclusion_proof.verify(&value, root).unwrap();
 /// ```
 #[derive(Debug, Default)]
-pub struct MerkleTree(CtMerkleTree<Sha256, Vec<u8>>);
+pub struct MerkleTree<T: AsRef<[u8]>>(CtMerkleTree<Sha256, T>);
```

Comment:
We are now using merkle trees with both `Vec<u8>` and `[u8; u32]`. It turns out that `CtMerkleTree` supports both, but unfortunately it has the trait bound on its type definitions. Maybe we should fork it and remove it.

Reply:
where are we using it with `[u8; u32]` for leaves?

Reply:
I see `[u8; 64]` (maybe you meant that)

Reply:
Yeah, I meant 64 bytes. Thanks for spotting!